### ISSUE TITLE
Pixel 6 cant add some widgets

### ISSUE THREAD

**Issue by GeoDerp** [iamrasel/lunar-launcher#88]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/iamrasel/lunar-launcher/issues) or [closed](https://github.com/iamrasel/lunar-launcher/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.

### App version

2.7.2

### Android version

13

### Steps to reproduce the bug

- swiped to left page 
- held down screen, until I saw 'Add Widget'
- selected gmail widget
- selected gmail account 
- selected inbox 
- widget setting jumps back to start, creating a loop (select gmail account)

I'm wondering if this bug occurs only with widgets that require selecting options before adding widget, This seems to happen on multiple apps.

### Expected behavior

widget added

**Comment by iamrasel** [comment]
Yes, this only happens to those widgets that need configuring before adding them. It happens because I added the widget support feature following the old implementation method, which doesn't support the above for some security reasons anymore.

**Comment by SkullGamer205** [comment]
Same problem in _Samsung Galaxy A31_
Some widgets can't add.
